pkg/provider/infisical/client.go

```diff
@@ -0,0 +1,128 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or impliec.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package infisical
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+
+	esv1beta1 "github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	"github.com/external-secrets/external-secrets/pkg/find"
+	"github.com/external-secrets/external-secrets/pkg/provider/infisical/api"
+)
+
+var errNotImplemented = errors.New("not implemented")
+
+// if GetSecret returns an error with type NoSecretError.
+// then the secret entry will be deleted depending on the deletionPolicy.
+func (p *Provider) GetSecret(ctx context.Context, ref esv1beta1.ExternalSecretDataRemoteRef) ([]byte, error) {
+	secret, err := p.apiClient.GetSecretByKeyV3(api.GetSecretByKeyV3Request{
+		EnvironmentSlug: p.apiScope.EnvironmentSlug,
+		ProjectSlug:     p.apiScope.ProjectSlug,
+		SecretPath:      p.apiScope.SecretPath,
+		SecretKey:       ref.Key,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return []byte(secret), nil
+}
```

Comment:
Is it expected to not support `property` on queries? If so, this should properly error out whenever it is set on `ref`

Reply:
| Used to select a specific property of the Provider value (if a map), if supported

Can you tell me what does property do. I am bit confused. Is this when you have a JSON key and extract it from there or something like that sort of?

Reply:
for `GetSecret` if the key is a JSON, it is a way to provide extraction from a json key within it - other providers implement this with `gjson`.

So a secret with:
```
{ 
 "foo":{"nested":"bar"}
 }
 ```
 could set:
 ```
 property: "" # returns {"foo":{"nested":"bar"}}
 property: "foo" #returns {"nested":"bar"}
 property: "foo.nested" #returns "bar"
 ```
 
 Same logic applies in `GetSecretMap`, with the difference that we Unpack the JSON object pointed to property:
 ```
 property: "" #returns foo: {"nested:"bar"}
property: "foo" # returns nested: bar
```

Reply:
Ahh makes sense. Will add support for it